File: trackr-app/app/src/main/java/com/io/trackr/trackr_app/entities/Anniversary.java
package entities;

import java.time.LocalDate;

public class Anniversary extends Event{
    /**
     * Create a new Anniversary event with a reminder time.
     * @param person The person this anniversary is associated with
     * @param date the date this event will take place
     * @param reminderDeadline when this event should be reminded to the user.
     */
    public Anniversary(Person person, LocalDate date, LocalDate reminderDeadline) {
        super(person, date, reminderDeadline);
    }

    /**
     * Create a new Anniversary event without a reminder time.
     * @param person The person this anniversary is associated with
     * @param date the date this event will take place
     */
    public Anniversary(Person person, LocalDate date) {
        this(person, date, null);
    }
}

File: trackrapp/app/src/main/java/com/trackr/trackr_app/entities/Anniversary.kt
package entities

import java.time.LocalDate

class Anniversary
/**
 * Create a new Anniversary event with a reminder time.
 * @param person The person this anniversary is associated with
 * @param date the date this event will take place
 * @param reminderDeadline when this event should be reminded to the user.
 */
    (person: Person?, date: LocalDate?, reminderDeadline: LocalDate?) : Event(person, date, reminderDeadline) {
    /**
     * Create a new Anniversary event without a reminder time.
     * @param person The person this anniversary is associated with
     * @param date the date this event will take place
     */
    constructor(person: Person?, date: LocalDate?) : this(person, date, null) {}
}